util/loaders/HeroLoader.ts

```diff
@@ -29,9 +29,37 @@ export class HeroLoader extends Loader<HeroMap> {
   }
 
   load() {
-    return this.generateHeroesMap();
+    const heroMap = this.generateHeroesMap();
+    return this.addRelatedBonds(heroMap);
   }
 
+  private addRelatedBonds = (heroMap: HeroMap) => {
+    const heroes = Object.keys(heroMap);
+    for (const hero of heroes) {
+      const bond4 = heroMap[hero].bondRequirments?.bond4;
+      const bond5 = heroMap[hero].bondRequirments?.bond5;
+      for (const relatedHero of heroes) {
+        if (bond4?.toLocaleLowerCase().includes(relatedHero.toLowerCase())) {
+          heroMap[relatedHero].relatedBonds.push({
+            prettyName: heroMap[hero].prettyName,
+            name: heroMap[hero].name,
+            text: bond4,
+            type: "Bond 4",
+          });
+        }
+        if (bond5?.toLocaleLowerCase().includes(relatedHero.toLowerCase())) {
+          heroMap[relatedHero].relatedBonds.push({
+            prettyName: heroMap[hero].prettyName,
+            name: heroMap[hero].name,
+            text: bond5,
+            type: "Bond 5",
+          });
+        }
+      }
+    }
+    return heroMap;
+  };
+
```

Comment:
This seems a little overcomplicated

`
  private addRelatedBonds = (heroMap: HeroMap) => {
    const heroes = Object.values(heroMap);
    for (const hero of heroes) {
      const bond4HeroName = hero.bondRequirments?.bond4Char;
      const bond5HeroName = hero.bondRequirments?.bond5Char;
      const bond4Hero = heroMap[bond4HeroName];
      const bond5Hero = heroMap[bond5HeroName];
      bond4Hero?.relatedBonds.push({
        prettyName: hero.prettyName,
        name: hero.name,
        text: hero.bondRequirments?.bond4,
        type: "Bond 4",
      });
      bond5Hero?.relatedBonds.push({
        prettyName: hero.prettyName,
        name: hero.name,
        text: hero.bondRequirments?.bond5,
        type: "Bond 5",
      });
    }
    return heroMap;
  }
`

bond4Char and bond5Char are values in the DB that need to be added to the loader and the bondReqs base interface